File: libs/shared/models/kotlin/AICoach.kt
package com.sentinel.shared.models

/**
 * Shared Kotlin data classes for AI coach suggestions.
 */

data class AISuggestionBase(
    val exercise_id: Int,
    val exercise_name: String,
    val current_weight: Double,
    val suggested_weight: Double,
    val reason: String,
    val confidence: Double
)

data class AISuggestionResponse(
    val exercise_id: Int,
    val exercise_name: String,
    val current_weight: Double,
    val suggested_weight: Double,
    val reason: String,
    val confidence: Double
)

File: libs/shared/models/typescript/ai-coach.ts
/**
 * Shared TypeScript interfaces for AI coach suggestions.
 */

export interface AISuggestionBase {
  exercise_id: number;
  exercise_name: string;
  current_weight: number;
  suggested_weight: number;
  reason: string;
  confidence: number;
}

export interface AISuggestionResponse extends AISuggestionBase {}
